app/Controllers/authController.php

```diff
@@ -122,7 +124,7 @@ public function formLoginAction() {
 			$username = Minz_Request::param('username', '');
 			$challenge = Minz_Request::param('challenge', '');
 
-			usleep(rand(100, 10000));	//Primitive mitigation of timing attacks, in μs
+			usleep(random_int(100, 10000));	//Primitive mitigation of timing attacks, in μs
```

Comment:
I think the much slower `random_int` is overkill here. We do not need a cryptographically secure value here

Reply:
after tests

```php
<?php
$start = microtime(true);
$sum = 0.0;
for ($i = 0; $i < 1000000; $i++) {
    $sum += rand(0, 32767);
}
printf('[rand] Time: %.3f s%s', microtime(true) - $start, PHP_EOL);

$start = microtime(true);
$sum = 0.0;
for ($i = 0; $i < 1000000; $i++) {
    $sum += mt_rand(0, 32767);
}
printf('[mt_rand] Time: %.3f s%s', microtime(true) - $start, PHP_EOL);

$start = microtime(true);
$sum = 0.0;
for ($i = 0; $i < 1000000; $i++) {
    $sum += random_int(0, 32767);
}
printf('[random_int] Time: %.3f s%s', microtime(true) - $start, PHP_EOL);

```
## result in PHP 8.1
```
[rand] Time: 0.044 s
[mt_rand] Time: 0.074 s
[random_int] Time: 0.732 s
```

## result in PHP 7.2
```
[rand] Time: 0.088 s
[mt_rand] Time: 0.071 s
[random_int] Time: 0.641 s
```
All these results are for 1 million iterations which is not our case we use it **only once**. The performances are therefore acceptable and of better quality